module_auto_update/models/module.py

```diff
@@ -41,31 +41,21 @@ def _compute_checksum_dir(self):
                     "Cannot compute dir hash for %s, module not found",
                     r.display_name)
 
+    @api.multi
     def _store_checksum_installed(self, vals):
-        if self.env.context.get('retain_checksum_installed'):
-            return
+        """Store the right installed checksum, if addon is installed."""
         if 'checksum_installed' not in vals:
-            if vals.get('state') == 'installed':
-                for r in self:
-                    r.checksum_installed = r.checksum_dir
-            elif vals.get('state') == 'uninstalled':
+            try:
```

Comment:
I prefer the other approach:
```python
version = vals.get('latest_version')
if version is None:
    return
```

Reply:
I also prefer EAFP

Reply:
I don't and I won't put it on any of my code, but I won't block this time.

Reply:
Developer preference, but this exact use case is noted in the [Python docs](https://docs.python.org/2/glossary.html#term-eafp) so I don't feel that you can ever block for this without good reason.

I'm also pretty sure the py2to3 converts 

```
try:
    ...
except X:
    pass
```

to

```
with contextlib.suppress(X):
    ...
```

It will not do the same for the LBFYL, which then becomes more lines and more resources.

Reply:
The question is that I don't suggest `LBYL` (https://docs.python.org/2/glossary.html#term-lbyl), but this, which is not:
```
version = vals.get('latest_version')
if version is None:
    return
```

Reply:
Wouldn't this change the logic of the code? The `if version is False` looks pretty explicit, where a `None` would be falling into the `else`. Is this the intent @Yajo - were there any `None` when testing or?

Reply:
No, the logic is the same: if no key is found, `None` is obtained in `version`, returning in the next condition, which is the same as in the `except:` part

Reply:
This is true, however you are not considering there actually being a `None` in the `vals`. The logic as currently written will send us past the `KeyError` and into the `else` clause of the `if` because `None is not False`.

If we switch to the `.get` approach, we have no concept of whether the `None` came from the `vals` or was simply a missing key. In this instance, we immediately return instead of triggering the if/else below it.

Reply:
That's not possible in ORM: it will never be a `None` as returned value. Any way, although you program bad your code, there's no problem with that.

Reply:
Do we know where `vals` comes from off-hand? I personally don't, but am assuming it's from a create or write method. Why would ORM restrict what vals are entered into a dictionary?

Reply:
Why are you trying to get an impossible case for my code? It works very well and doesn't mean any error even on the improbable case of somebody consciously wants to ruin it, but hey, if you have a big problem with that, just leave it!

Reply:
I wasn't going for ruining it, I was just pointing out that this could have unintended side-effects outside of what we think. 

The `if version is False:` is an incredibly explicit if statement that makes me think @Yajo saw something in his testing where a False-type value can be in `vals` but not actually be a `False`. Maybe this is just a `"0"` version number; I'm not sure.

`None` is a False type value though - so while we may think it won't cause issues, this is why I asked @Yajo for clarification to the explicit if in https://github.com/OCA/server-tools/pull/948#discussion_r135314276

Honestly at this point I'd kind of like an inline comment to the edge case that the explicit if is covering. @Yajo can you clarify for us in a comment why we are being so explicit?

Reply:
I wasn't trying to start a flamewar here. It's very easy:

- Odoo stores `False` for empty fields, **always**, so no need to check if there's `None` or whatever.
- If above point is true always, [it's even more in current case](https://github.com/odoo/odoo/blob/15cb84e0f77db350560aac9d32d947e919def128/openerp/addons/base/module/module.py#L472).
- What would happen if a module is defined with `"version": ""`? This is very unlikely, and if it ever happened, I'm pretty sure Odoo would fail or patch it with `"9.0"` or `"10.0"` automatically, but... _is it even worth it testing that? IMHO not._

So, if I wanted to use LBYL, the right code would be:

```python
# Option A, matching 100% current behavior
if "latest_version" not in vals:
    return
# Option B, not 100%, but side matches would not matter since they should theoretically never happen
if vals.get("latest_version") is None:
    return
```

...  but I didn't.

The fastest way to do it was as it is right now, and it uses EAFP because IMHO it's the pythonic way to do it.

However, there's one basic truth that we all have to acknowledge: **Sometimes there's more than 1 way to do the same thing right**. So, in case somebody else chooses a way that you wouldn't choose by yourself, but it's also a good way to get to the exact same result where the only difference is a matter of taste... why bother? **Hakuna matata!**

----

As a side note, kudos @lasley for finding out https://github.com/OCA/server-tools/pull/948#discussion_r135288685, a very nice addition to Python 3. I'm very eager to get it! ☺️

Reply:
https://github.com/OCA/server-tools/pull/948#discussion_r135288685 is another sign that they have found EAFP very cluttered in code :wink: but hey, as you have said, there are several ways of doing things. You won't see me a lot doing EAFP though, not at least on that form.